libs/ui/src/change_precinct_button.tsx

```diff
@@ -81,6 +90,11 @@ export function ChangePrecinctButton({
     const { value } = event.currentTarget;
     if (!value) return; // in case of blur on placeholder option
 
+    if (value === dropdownCurrentValue) {
+      // blur on current value should be no-op
+      return;
+    }
+
```

Comment:
It was firing twice in some (but not all) tests, once on click and again on blur.